tests/Feature/RegisterTest.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+namespace Tests\Feature;
+
+use Tests\TestCase;
+use App\Models\User;
+use Firebase\JWT\JWT;
+use App\Traits\InteractsWithJWT;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+
+class RegisterTest extends TestCase
+{
+    use RefreshDatabase, InteractsWithJWT;
+
+    /**
+     * @param  array  $override
+     * @return array
+     */
+    private function validParamaters($override = [])
+    {
+        return array_merge([
+            'name' => 'Test User Name',
+            'email' => 'test@example.org',
+            'password' => 'password',
+            'password_confirmation' => 'password',
+        ], $override);
+    }
+
+    /** @test */
+    public function a_guests_can_register_for_an_account()
+    {
+        $this->postJson(route('auth.register'), $this->validParamaters([
+            'name' => 'Test User Name',
+            'email' => 'test@example.org',
+        ]))->assertStatus(200);
+
+        $user = User::first();
+        $this->assertEquals('Test User Name', $user->name);
+        $this->assertEquals('test@example.org', $user->email);
+    }
+
+    /** @test */
+    public function a_valid_token_is_returned_when_the_user_registers()
```

Comment:
Can we merge this with the test above?

Reply:
Sure thing. Updated test name to `a_guest_can_register_for_an_account_and_get_a_valid_token`